pkg/descriptor/parse.go

```diff
@@ -38,9 +43,9 @@ func GetPathFromExtractedTarGz(imageSpec types.ImageSpec, insecureRegistry bool)
 	// pull image layer
 	var layer v1.Layer
 	if insecureRegistry {
-		layer, err = crane.PullLayer(imageRef, crane.Insecure)
+		layer, err = crane.PullLayer(imageRef, crane.Insecure, crane.WithAuthFromKeychain(keyChain))
 	} else {
-		layer, err = crane.PullLayer(imageRef)
+		layer, err = crane.PullLayer(imageRef, crane.WithAuthFromKeychain(keyChain))
```

Comment:
I think this is not the only place where this is necessary. Also e.g. for the config layer this might become a problem.

Reply:
thanks for pointing out, add keyChain support to config layer